src/components/InputEditor.tsx

```diff
@@ -97,9 +98,15 @@ export default function InputEditor({ output }: Props) {
     const encodedInput = url.searchParams.get('code')
     const storedInput = localStorage.getItem(STORAGE_KEY)
     if (encodedInput) {
-      setCode(ungzip(Base64.toUint8Array(encodedInput), { to: 'string' }))
+      setDeferredCode(ungzip(Base64.toUint8Array(encodedInput), { to: 'string' }))
```

Comment:
`useEffect` is already executed asynchronously, not during the render phase. It's not worth to do that because, as you see, the code looks complex, and the same logic is written twice.